indexer_app/tasks.py

```
@@ -25,35 +36,61 @@ async def indexer(from_block: int, to_block: int):
+            )
+
+            # Log total time for one iteration
+            iteration_end_time = time.time()
+            logger.info(
+                f"Total time for one iteration: {iteration_end_time - fetch_start_time:.4f} seconds"
+            )
+        
+
+        except asyncio.TimeoutError:
+            logger.warning("Stream stalled: no new blocks within timeout, restarting...") # raise Exception so sytemd can restart the worker
+            raise Exception("Stream stalled: restarting...")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)